domain/src/main/java/joist/domain/AbstractQueries.java

```diff
@@ -29,6 +32,38 @@ public T find(Long id) {
     return UoW.load(this.domainType, id);
   }
 
+  public List<T> find(Long... ids) {
+    // Use load as it hits the IdentityMap and could avoid an unneeded query
+    return UoW.load(this.domainType, Arrays.asList(ids));
+  }
+
+  public List<T> find(Collection<Long> ids) {
+    // Use load as it hits the IdentityMap and could avoid an unneeded query
+    return UoW.load(this.domainType, ids);
+  }
+
+  public Iterator<List<T>> findInBatches(Integer batchSize) {
+    AbstractQueries<T> queries = this;
+    Iterator<List<T>> it = new Iterator<List<T>>() {
+
+      private Integer offset = 0;
+      private int total = (int) queries.count();
+
+      @Override
+      public boolean hasNext() {
+        return this.offset < this.total;
+      }
+
+      @Override
+      public List<T> next() {
+        List<Long> ids = Select.from(queries.aliasType).limit(batchSize).offset(this.offset).listIds();
+        this.offset += batchSize;
+        return queries.find(ids);
```

Comment:
This makes sense, but we're not actively removing entities from the IdentityMap, so if we load things in batches of 5000, then the first 5000 works, next 5000 works, but next 5000 blows up because the IdentityMap has 10k objects in it.

So, I'm tempted to think that unless we solve that, and really have a way to scroll through ~X objects at a time, and flush/evict them from the IdentityMap as we go, the effective max objects is still ~10k, and at that point loading them in batches probably doesn't matter.

If you want to split this findInBatches out as a separate pull request while we mull/kibitz about that, the find(ids) methods both look good and ready to go.